### ISSUE TITLE
Help,how can I get listeners by event name in v3?

### ISSUE THREAD

**Issue by Moln** [zendframework/zend-eventmanager#25]
When I use `EM` V2, I have a code in my project for implement `isLastListener`.
How can I implement in V3?
Does there have a solution?

``` php
use Zend\EventManager\AbstractListenerAggregate;
use Zend\EventManager\Event;
use Zend\EventManager\EventManagerInterface;

class TestListener extends AbstractListenerAggregate
{
    /** @var  EventManagerInterface */
    protected $events;

    public function isLastListener()
    {
        $listeners = $this->events->getListeners('event.name');
        $listeners = iterator_to_array($listeners);
        if (empty($listeners)) {
            return false;
        }

        $last = end($listeners);
        $call = $last->getCallback();

        return is_array($call) && $call[0] == $this;
    }

    public function attach(EventManagerInterface $events, $priority = 1)
    {
        $this->events = $events;
        $this->listeners[] = $lis = $events->attach('event.name', [$this, 'test']);
    }

    public function test(Event $e)
    {
        if ($this->isLastListener()) {
            //If is last listener, to do something.
        }
    }
}
```

I saw v3 code. Can change private `getListenersByEventName` to public?

**Comment by weierophinney** [comment]
> Can change private getListenersByEventName to public?

No; we explicitly removed this use case from the public API.

One reason is because there are huge differences from v2 in how listeners are aggregated internally; keeping the method would have broken existing code in subtle, hard-to-fix ways, and providing a migration path would have been impossible. (Related: we actually went through 3 different iterations of aggregation before pinning the stable v3 release; having to change test expectations for each iteration was going to be a nightmare, so we removed them before the second iteration.)

The primary reason, though, is because it was a very infrequently used feature with a high maintenance burden. The number of use cases for it are quite slim, and the primary use case — retrieving `CallbackHandler` instances to allow detaching listeners — went away when we stopped wrapping callables in `CallbackHandler` instances.

Looking at your above code, there's a much better approach possible that will work with both v2 and v3 already. Register your listener at a really, really low negative priority to ensure it executes last. A good bet is that nobody is listening at `- PHP_INT_MAX`.
